### ISSUE TITLE
Does not support strict mode?

### ISSUE THREAD

**Issue by LukasGerm** [flackr/scroll-timeline#119]
Hello everyone, 

I made this small pen https://codepen.io/lukasgerm/pen/oNQJbWa which is using the polyfill. But the SDK is throwing an error 

```
ReferenceError: assignment to undeclared variable parts
```

It is probably related to codepen enabling strict mode by or so (just an assumption). What do you guys suggest here?

**Comment by LukasGerm** [comment]
When I try to rebuild it to use the javascript api I always get 

```
Uncaught TypeError: setting getter-only property "axis"
```

With this code

```javascript
const cards = document.getElementsByClassName("card")

for(let i = 0; i < cards.length; i++) {
    const timeline = new ViewTimeline({axis:"x"});
  const keyframes = [
    {
      scale: 0.85,
      offset: 0.2
    },
    {
      scale: 1,
      offset: 0.5
    },
    {
      scale: 0.85,
      offset: 0.8
    },
  ]
  cards[i].animate(keyframes,{timeline})
};
```

**Comment by brogr** [comment]
> When I try to rebuild it to use the javascript api I always get
> 
> ```
> Uncaught TypeError: setting getter-only property "axis"
> ```
> 
> With this code
> 
> ```js
> const cards = document.getElementsByClassName("card")
> 
> for(let i = 0; i < cards.length; i++) {
>     const timeline = new ViewTimeline({axis:"x"});
>   const keyframes = [
>     {
>       scale: 0.85,
>       offset: 0.2
>     },
>     {
>       scale: 1,
>       offset: 0.5
>     },
>     {
>       scale: 0.85,
>       offset: 0.8
>     },
>   ]
>   cards[i].animate(keyframes,{timeline})
> };
> ```

Same issue here!

**Comment by LukasGerm** [comment]
I can confirm that it works again when I roll back to this commit as well.

**Comment by bramus** [comment]
This issue seems to enclose two different issues.

The [first message in this thread](https://github.com/flackr/scroll-timeline/issues/119#issue-1828641157) is about strict mode and the `parts` variable. I’ve got a PR up to fix this: #136.

[Further down in this thread](https://github.com/flackr/scroll-timeline/issues/119#issuecomment-1658247473) there is a mention of https://codepen.io/lukasgerm/pen/WNYLGYR which shows a different bug: it throws `TypeError: null is not an object (evaluating 'n.rangeName')`. Digging down that demo, it is because no range information is passed into the options. PR #137 fixes that.

I’ve not been able to reproduce the issue about `axis` which is mentioned in [this comment](https://github.com/flackr/scroll-timeline/issues/119#issuecomment-1658197563).
